docs/source/onnx_dynamo.rst

```diff
@@ -0,0 +1,146 @@
+TorchDynamo-based ONNX Exporter
+===============================
+
+.. contents:: :local:
+    :depth: 3
+
+.. warning::
+  The ONNX exporter for TorchDynamo is under active development and is subject to rapid change.
+
+Overview
+--------
+
+As mentioned before, the ONNX exporter leverages TorchDynamo engine to hook into Python's frame evaluation API
+and dynamically rewrite its bytecode into an FX Graph.
+The resulting FX Graph is then polished before it is finally translated into an ONNX graph.
+
+The main advantage of this approach is that the `FX graph <https://pytorch.org/docs/stable/fx.html>_` is captured using
+bytecode analysis that preserves the dynamic nature of the model instead of using traditional static tracing techniques.
+
+The exporter is designed to be modular and extensible. It is composed of the following components:
+
+  - **ONNX Registry**: A registry of ONNX operators and functions.
+  - **ONNX Exporter**: The main exporter that orchestrates the export process.
+  - **ONNX Export Output**: The output of the exporter that contains the exported ONNX graph and diagnostics.
+  - **ONNX Export Output Serializer**: A serializer that serializes the export output to a file.
+  - **ONNX Diagnostic Options**: A set of options that control the diagnostics emitted by the exporter.
+  - **ONNX Export Options**: A set of options that control the export process.
+
+Dependencies
+------------
+
+The ONNX exporter depends on extra Python packages not available through PyTorchs's `requirements.txt` file:
+
+  - `ONNX <https://onnx.ai>`_
+  - `ONNX Script <https://onnxscript.ai>`_
+
+They can be installed through `pip <https://pypi.org/project/pip/>`_ utility as follows:
+
+.. code-block:: bash
+
+  pip install onnx onnxscript-preview    # TODO: Replace with `onnxscript` when it is official
+
+A simple example
+----------------
+
+See below a demonstration of exporter API in action with a simple image classifier model as example:
+
+.. code-block:: python
+
+  import torch
+
+  class MNISTModel(torch.nn.Module):
```

Comment:
I am wondering if we should use MLP as the first example.

Reply:
What would change? The ultimate goal for the first example is show how to use the public API. The actual model architecture is kind of secondary given the promise of "work (in almost) every model"

Reply:
Yep. The first example must be as simple as possible. This is ALWAYS a problem in Office and a lot of our ML softwares.